server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -2429,20 +2446,20 @@ public Pair<List<? extends IpAddress>, Integer> searchForIPAddresses(final ListP
         final Long networkId = cmd.getNetworkId();
         final Long vpcId = cmd.getVpcId();
 
-        final String state = cmd.getState();
+        final List<IpAddress.State> states = getStatesForIpAddressSearch(cmd);
         Boolean isAllocated = cmd.isAllocatedOnly();
         if (isAllocated == null) {
-            if (state != null && (state.equalsIgnoreCase(IpAddress.State.Free.name()) || state.equalsIgnoreCase(IpAddress.State.Reserved.name()))) {
+            if (states.contains(IpAddress.State.Free) || states.contains(IpAddress.State.Reserved)) {
```

Comment:
we cannot have a null here, so no biggy but maybe reverse the checks and even use `Arrays`?